File: public/programs/c/p11.c
// Check whether is a palindrome or not

#include<stdio.h>

int main()
{
	int n,r=0,k;
	printf("Enter a number");
	scanf("%d",&n);
	
	k=n;                           // assigning n value to k as logic involves alteration of n's value
	while(k!=0)
	{
		r = r*10;                             
		r = r + k%10;                         
		k = k /10;
	}
	
	if(n==r)
	{
		printf("%d is a palindrome",n);
	}
	else{
		printf("%d is not a palindrome",n);
	}
}

		

File: public/programs/java/p11.java
import java.util.*;
class p11
{
	public static void main(String[] args) {
		int a,rev=0,r,b;
		Scanner scan=new Scanner(System.in);
		System.out.print("Enter a number: ");
		a=scan.nextInt();
		b=a;
		while(a>0)
		{
            r=a%10;
            rev=rev*10+r;
            a=a/10;
		}
		if(b==rev)
			System.out.print(b +" is a palindrome");
		else
			System.out.print(b +" is not a palindrome");

	}
}